app/models/piece.rb

```diff
@@ -2,5 +2,47 @@ class Piece < ActiveRecord::Base
   
   belongs_to :game
   belongs_to :player
+
+  def is_obstructed?(destination_x, destination_y)
+    location_x = self.x_position
+    location_y = self.y_position
+    if location_x == destination_x
+      # check for vertical obstruction
+      location_y > destination_y ? incrementer = -1 : incrementer = 1
+      position_y = location_y + incrementer
+      while position_y != destination_y
+        if game.pieces.where(x_position: location_x, y_position: position_y).any?
```

Comment:
@lasley I like the idea and played around with it, but there are complications with reverse ranges. When the piece is moving down or to the left, you get a reverse range like 4...1 and it seems like Ruby doesn't like those: http://stackoverflow.com/questions/2070574/is-there-a-reason-that-we-cannot-iterate-on-reverse-range-in-ruby